src/DynamoCore/Core/PulseMaker.cs

```diff
@@ -0,0 +1,152 @@
+﻿using System;
+using System.Linq;
+using System.Threading;
+
+using Dynamo.Core.Threading;
+using Dynamo.Models;
+
+namespace Dynamo.Core
+{
+    public class PulseMaker
```

Comment:
Also, `PulseMaker` feels way too general for a component that does something so specific.  Why not call this something that alludes to the fact that it causes runs?